snapcraft/plugins/kernel.py

```diff
@@ -179,9 +180,33 @@ def _unpack_generic_initrd(self):
                 'unsquashfs', self.os_snap, os.path.dirname(initrd_path)],
                 cwd=temp_dir)
 
+            tmp_initrd_path = os.path.join(
+                temp_dir, 'squashfs-root', initrd_path)
+
+            mime_detector = magic.open(
+                magic.MAGIC_MIME_TYPE | magic.MAGIC_ERROR)
+            mime_detector.load()
+            mime_type = mime_detector.file(tmp_initrd_path)
+            if not mime_type:
+                raise RuntimeError(
+                    'Unable to determine mime type for {!r}: {}'.format(
+                        tmp_initrd_path, os.strerror(mime_detector.errno())))
+            logger.debug('initrd mime_type: {} {}'.format(
+                tmp_initrd_path, mime_type))
+
+            # Support gzip (can be gzip or x-gzip)
+            if 'gzip' in mime_type:
```

Comment:
I think this check should be explicit for gzip or x-gzip.
There could be a case where we find foo-gzip, and we should fail with the unsupported exception.